test/integration/router_test.go

```diff
@@ -562,14 +618,16 @@ func TestRouterPathSpecificity(t *testing.T) {
 	// same service, and that that service varies its response in accordance with
 	// the path, so we still get the tr.HelloPodPath response when we request
 	// /test even though we request using routeAddress.
-	if err := waitForRoute(routeTestAddress, "www.example.com", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	if err := waitForRoute(routeAddress, "www.example.com", "http", nil, tr.HelloPod); err != nil {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	if err := waitForRoute(routeTestAddress, "www.example.com:80", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
+	for _, proto := range protocols {
+		if err := waitForRoute(routeTestAddress, "www.example.com", proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
+		if err := waitForRoute(routeAddress, "www.example.com", proto.name, nil, tr.HelloPod); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
+		if err := waitForRoute(routeTestAddress, "www.example.com:" + proto.port, proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
```

Comment:
same comments as earlier re: adding protocol information to the failure messages (the 3 `t.Fatalf`s in the for loop).